src/main.rs

```diff
@@ -36,11 +44,19 @@ fn main() {
     }
 
     // Create handler to deal with HTTP requests
-    let http_request_handler = HttpRequestHandler::new();
+    let http_request_handler = Arc::new(HttpRequestHandler::new());
+
+    // Create a heartbeat thread for the ComponentManager
+    let heartbeat_handler_ref = http_request_handler.clone();
```

Comment:
Why do you need to clone the handler?

Reply:
It’s an reference counter ptr. It’s a little confusing, but cloning here just gives us another handle for the other thread. Perhaps this should be a weak reference?